src/validation.cpp

```diff
@@ -4181,8 +4181,8 @@ void CChainState::CheckBlockIndex()
 
     // Build forward-pointing map of the entire block tree.
     std::multimap<CBlockIndex*,CBlockIndex*> forward;
-    for (const std::pair<const uint256, CBlockIndex*>& entry : m_blockman.m_block_index) {
-        forward.insert(std::make_pair(entry.second->pprev, entry.second));
+    for (std::pair<const uint256, CBlockIndex>& entry : m_blockman.m_block_index) {
+        forward.insert(std::make_pair(entry.second.pprev, &entry.second));
```

Comment:
nit: (assuming it compiles)

```suggestion
    for (const auto& [_, block] : m_blockman.m_block_index) {
        forward.emplace(block.pprev, &block));
```